### ISSUE TITLE
FR: Make moving more visually intuitive

### ISSUE THREAD

**Issue by imeed166** [artem-barmin/obsidian-block-drag-n-drop#2]
1. **Smart move :** As moving one type of block to the other is extremely rare, removing the lines and only allowing moves between the same types of block will keep the user from making mistakes :
    - when moving list, show line between other list item only, not between paragraph or headings ;
    - when moving headings, show line between headings only, not between paragraph or lists items ;
    - when moving paragraphs, show line between paragraphs and heading, but not between list items.
2. **Indentation help :** When moving list items between indentation levels, make the line length adapt to the destination to show the user at which level he is putting the list items ;
3. **Disable cursor :** Keep only the line destination and disable the cursor when moving block as we only need to move **between** blocks, not **before**, in the **middle** or **after** blocks (in the same line) ;
4. **Visual help :** When dragging the dots in the left, display the content you're moving in a box, to help the user understand what is included in the move.

**Comment by artem-barmin** [comment]
Great suggestions! 1 and 3 is easily doable, added them to roadmap. Can you explain in picture point 2? Point 4 is doable, but complex, will think about them

**Comment by imeed166** [comment]
Here is the equivalent of point 2 in notion

https://user-images.githubusercontent.com/38974541/163573879-e9534d8f-8bb3-4d07-9955-1208a0b29770.mov

**Comment by imeed166** [comment]
Point 4 is already implemented in this plugin https://github.com/GitMurf/obsidian-drag-and-drop-blocks, maybe it'll help.

**Comment by artem-barmin** [comment]
Point 2 is clear. Will add it to roadmap. BTW haven't seen this plugin, thank you for reference!

**Comment by imeed166** [comment]
The plugin only works with the legacy editor.

**Comment by artem-barmin** [comment]
@imeed166 published new version and update demo, check it out

**Comment by imeed166** [comment]
I couldn't try it because it was slow compared to last version

https://user-images.githubusercontent.com/38974541/163665034-ef8c3df1-c614-4686-bf36-816fc6690c47.mov

**Comment by artem-barmin** [comment]
@imeed166 New release: added caching, by measures - now it's much faster, but visually for it's hard to check because of fast cpu. Please review

**Comment by imeed166** [comment]
It's faster, but still slow compared to 1.0.0

https://user-images.githubusercontent.com/38974541/163675268-afa31db7-9fed-413e-a35e-c1320704b853.mov

**Comment by artem-barmin** [comment]
Ok, got it. I have idea how to speedup, but it will take some time to rewrite

**Comment by artem-barmin** [comment]
@imeed166 One more try. Rewritten list item handling fully based on Obsidian cache, added throttle for dragover handler. It will fire only once in 10ms, should not overload computer with consecutive executions. 

Please also add OS/CPU/RAM/Obsidian version info that you use

**Comment by imeed166** [comment]
A bit better

https://user-images.githubusercontent.com/38974541/163715185-c181989c-0833-49f7-8c14-e732252c3b59.mov


Obsidian : 0.14.5 / Installer 0.14.5
macOS 12.1
MacBook Pro, early 2015
Processor : 2,7 GHz Dual-Core Intel Core i5
Memory : 8 GB 1867 MHz DDR3

**Comment by artem-barmin** [comment]
@imeed166 Try one more time. Refactored dragover handling

**Comment by imeed166** [comment]
It's fast again now, thank you. I have two more issues now (don't know if I need to open another thread or not)
1. I can't find a reorder list items. Whenever a drop an item, it's becomes a sub-item
2. When moving an item, it appears briefly at the bottom before the animation starts (see video)


https://user-images.githubusercontent.com/38974541/163823161-7a5604af-08fa-4c2d-9c72-8f2954edc41e.mov

**Comment by artem-barmin** [comment]
1. There is no way to reorder items, and frankly speaking it's even not in the roadmap, because main use case for me - fast creation of embeds in other panes.
2. That's the way how "dragged content" is prepared. I can't make it transparent because it will not be visible during dragging. At my computer it's not noticeable at all. At slower computer unfortunately - I don't know how to fix this

**Comment by artem-barmin** [comment]
But you can open another issue regarding reordering - maybe someday I will have time to fix this

**Comment by artem-barmin** [comment]
But point 4 already implemented, at least as I understand it. Dragged content is visualised near cursor, I can even see it on your last video

**Comment by imeed166** [comment]
Ah right, sorry, didn't notice.
